src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py

```diff
@@ -285,42 +323,74 @@ def log_artifact(self, artifact_folder: str) -> None:
         if not os.listdir(artifact_folder):
             LOGGER.error("The path to the artifact is empty.")
             return
+        if not os.path.isfile(os.path.join(artifact_folder, EvalRun.EVALUATION_ARTIFACT)):
+            LOGGER.error("The run results file was not found, skipping artifacts upload.")
+            return
         # First we will list the files and the appropriate remote paths for them.
-        upload_path = os.path.basename(os.path.normpath(artifact_folder))
-        remote_paths = {"paths": []}
+        root_upload_path = posixpath.join("promptflow", 'PromptFlowArtifacts', self.info.run_name)
+        remote_paths = {'paths': []}
         local_paths = []
-
+        # Go over the artifact folder and upload all artifacts.
         for (root, _, filenames) in os.walk(artifact_folder):

```

Comment:
We are not expecting it to have more than one file right ?

Reply:
Technically yes, however, I wanted to make this code more universal for future use.